editor/components/rich-text/tinymce.js

```diff
@@ -9,15 +9,101 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { Component, createElement } from '@wordpress/element';
+import { keycodes } from '@wordpress/utils';
 
 /**
  * Internal dependencies
  */
 import { diffAriaProps, pickAriaProps } from './aria';
 import { valueToString } from './format';
 
+const { BACKSPACE, DELETE } = keycodes;
+
+/**
+ * Determines whether we need a fix to provide `input` events for contenteditable.
+ *
+ * @param {Element} editorNode The root editor node.
+ *
+ * @return {boolean} A boolean indicating whether the fix is needed.
+ */
+function needsInternetExplorerInputFix( editorNode ) {
+	return (
+		// Rely on userAgent in the absence of a reasonable feature test for contenteditable `input` events.
+		/Trident/.test( window.navigator.userAgent ) &&
```

Comment:
Could we borrow what Modernizr does here? Or is that for a different `input` event?

https://github.com/Modernizr/Modernizr/blob/master/feature-detects/event/oninput.js

Reply:
I thought about that too. The test returns true for even a non-contenteditable `<div>` in IE11 because of the first part that sets an `oninput` attribute and checks the type of the element's `oninput` property. The browser _does_ support the `input` event but doesn't dispatch it for `contenteditable` text input.

I tried to adapt the latter half of the test that dispatches a KeyboardEvent and listens for `input` but didn't have any luck getting it to work in IE11. It's possible I missed something but didn't think I should spend more time on it.

Reply:
Also, thank you for taking the time to look and make this suggestion, @noisysocks . I actually forgot that I'd looked at this until I worked through it again. :)